# Westlake Realty Group Proposal

Authors: Drew Holcombe, Juan Acosta

## Overview
Our project presents data of house sale prices for King County, Washington. Using statistical analyses techniques in Python we are able to give our recomendations on which factors of a hosue are the most significant to determine price, and how Westlake Realty Group could use this information to reduce risk and increase profit.


## Business Challenge

Westlake Realty Group (hypothetical example) is looking for recomendations on factors and carateristics of a house that are the most impactful to determine its price. Our group's objective is to find factors that accurately predict the given target, driven by real world data from the housing market, and provide results and justification for them to consider on their decision making process.


## Data Understanding
Our project presents data of house sale prices for King County, Seattle, contained in the kc_house dataset. It includes houses sold between May 2014 and May 2015. The variables we found to be most useful are --
We used 


In [2]:
import numpy as np
import pandas as pd
import seaborn as sns
from matplotlib import pyplot as plt
from sklearn.preprocessing import PolynomialFeatures, StandardScaler, OneHotEncoder, OrdinalEncoder
from sklearn.linear_model import LinearRegression
import statsmodels.api as sm
from scipy import stats

In [7]:
df = pd.read_csv('data/kc_house_data.csv')
df.shape

(21597, 21)

We have information of 21597 houses located in King county washington

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21597 entries, 0 to 21596
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   id             21597 non-null  int64  
 1   date           21597 non-null  object 
 2   price          21597 non-null  float64
 3   bedrooms       21597 non-null  int64  
 4   bathrooms      21597 non-null  float64
 5   sqft_living    21597 non-null  int64  
 6   sqft_lot       21597 non-null  int64  
 7   floors         21597 non-null  float64
 8   waterfront     19221 non-null  object 
 9   view           21534 non-null  object 
 10  condition      21597 non-null  object 
 11  grade          21597 non-null  object 
 12  sqft_above     21597 non-null  int64  
 13  sqft_basement  21597 non-null  object 
 14  yr_built       21597 non-null  int64  
 15  yr_renovated   17755 non-null  float64
 16  zipcode        21597 non-null  int64  
 17  lat            21597 non-null  float64
 18  long  

There are null values present in 'waterfront', 'view' and 'yr_renovated' columns

## Data Preparation

We first clean the data, look for duplicate data points, identify noticible outliers, and droping null values 

In [ ]:
`id` - Unique identifier for a house

In [8]:
df['id'].value_counts()

795000620     3
1825069031    2
2019200220    2
7129304540    2
1781500435    2
             ..
7812801125    1
4364700875    1
3021059276    1
880000205     1
1777500160    1
Name: id, Length: 21420, dtype: int64